scipy/optimize/_minimize.py

```diff
@@ -190,21 +193,27 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
 
         For method-specific options, see :func:`show_options()`.
     callback : callable, optional
-        Called after each iteration. For 'trust-constr' it is a callable with
+        A callable called after each iteration.
+
+        All methods except TNC, SLSQP, and COBYLA support a callable with
         the signature:
 
-            ``callback(xk, OptimizeResult state) -> bool``
+            ``callback(*, OptimizeResult res_i) -> bool``
```

Comment:
`intermediate_result`

Reply:
What's the point in returning a `bool`? Is it still to halt execution?

Reply:
No. It does not need to return a book. That is a mistake.